### ISSUE TITLE
keras plot_model inserts strange first input row containing long integer number

### ISSUE THREAD

**Issue by sbrege** [keras-team/keras#11376]
Hi,

I tried to use keras plot_model to visualize my lstm-modell. I have installed the latest version of graphviz in my windows 8 machine. I also work with the latest anaconda 64-bit version.

```
# -*- coding: utf-8 -*-
import os
os.environ["PATH"] += os.pathsep + 'C:/Program Files (x86)/graphviz/bin/'
import pydot_ng as pydot
pydot.find_graphviz()
from keras.utils.vis_utils import plot_model
from keras.models import Sequential
from keras.layers import Dense, Embedding, LSTM, Dropout

embedding_vector_dimensionality = 256
max_words = 10000  # We will only consider the top 10,000 words in the dataset
embedding_dropout_factor = 0.5
recurrent_dropout_factor = 0.2
LSTM_dropout_factor = 0.2
number_LSTM_layer = 1
LSTM_output_neurons = 50
layer_dropout_factor = 0.2
    
model = Sequential()
model.add(Embedding(max_words, embedding_vector_dimensionality, input_length=400,
                    name='embed'))
model.add(Dropout(embedding_dropout_factor))
for layerNum in range(number_LSTM_layer):
    layerName = "layer{}".format(layerNum)
    print("Added LSTM_Layer {}".format(layerNum))
    model.add(LSTM(units=LSTM_output_neurons, return_sequences=True,
               recurrent_dropout=recurrent_dropout_factor,
               dropout=LSTM_dropout_factor))
    model.add(Dropout(layer_dropout_factor))
model.add(LSTM(units=LSTM_output_neurons, recurrent_dropout=recurrent_dropout_factor, dropout=LSTM_dropout_factor))
model.add(Dropout(layer_dropout_factor))
model.add(Dense(8, activation='sigmoid'))

plot_model(model, to_file='deployed_model.png', show_shapes=True, show_layer_names=True)
```

The Problem is, as you can see in the linked output, that the graph has a strange input-row, which contains a large number. This row is also not contained in my sequential model, even if I saved it as json-file with model.to_json(). There the first row is the embedding:

https://goo.gl/v2WgzL

How can I get rid of that row (other solution than removing it in a photoshop)?

Thank you!

sbrege

The output of conda list -e > requirements.txt is:
```
# This file may be used to create an environment using:
# $ conda create --name <env> --file <this file>
# platform: win-64
alabaster=0.7.11=py36_0
asn1crypto=0.24.0=py36_0
astroid=2.0.4=py36_0
babel=2.6.0=py36_0
backcall=0.1.0=py36_0
bleach=2.1.4=py36_0
ca-certificates=2018.03.07=0
certifi=2018.8.24=py36_1
cffi=1.11.5=py36h74b6da3_1
chardet=3.0.4=py36_1
cloudpickle=0.5.5=py36_0
colorama=0.3.9=py36h029ae33_0
cryptography=2.3.1=py36h74b6da3_0
decorator=4.3.0=py36_0
docutils=0.14=py36h6012d8f_0
entrypoints=0.2.3=py36_2
html5lib=1.0.1=py36_0
icu=58.2=ha66f8fd_1
idna=2.7=py36_0
imagesize=1.1.0=py36_0
ipykernel=4.9.0=py36_0
ipython=6.5.0=py36_0
ipython_genutils=0.2.0=py36h3c5d0ee_0
isort=4.3.4=py36_0
jedi=0.12.1=py36_0
jinja2=2.10=py36_0
jpeg=9b=hb83a4c4_2
jsonschema=2.6.0=py36h7636477_0
jupyter_client=5.2.3=py36_0
jupyter_core=4.4.0=py36_0
keyring=13.2.1=py36_0
lazy-object-proxy=1.3.1=py36hfa6e2cd_2
libpng=1.6.34=h79bbb47_0
libsodium=1.0.16=h9d3ae62_0
markupsafe=1.0=py36hfa6e2cd_1
mccabe=0.6.1=py36_1
mistune=0.8.3=py36hfa6e2cd_1
nbconvert=5.3.1=py36_0
nbformat=4.4.0=py36h3a5bc1b_0
numpydoc=0.8.0=py36_0
openssl=1.0.2p=hfa6e2cd_0
packaging=17.1=py36_0
pandoc=2.2.3.2=0
pandocfilters=1.4.2=py36_1
parso=0.3.1=py36_0
pickleshare=0.7.4=py36h9de030f_0
pip=10.0.1=py36_0
prompt_toolkit=1.0.15=py36h60b8f86_0
psutil=5.4.7=py36hfa6e2cd_0
pycodestyle=2.4.0=py36_0
pycparser=2.18=py36_1
pyflakes=2.0.0=py36_0
pygments=2.2.0=py36hb010967_0
pylint=2.1.1=py36_0
pyopenssl=18.0.0=py36_0
pyparsing=2.2.0=py36_1
pyqt=5.9.2=py36ha878b3d_0
pysocks=1.6.8=py36_0
python=3.6.6=hea74fb7_0
python-dateutil=2.7.3=py36_0
pytz=2018.5=py36_0
pywin32=223=py36hfa6e2cd_1
pyzmq=17.1.2=py36hfa6e2cd_0
qt=5.9.6=vc14h62aca36_0
qtawesome=0.4.4=py36h5aa48f6_0
qtconsole=4.4.1=py36_0
qtpy=1.5.0=py36_0
requests=2.19.1=py36_0
rope=0.11.0=py36_0
setuptools=40.2.0=py36_0
simplegeneric=0.8.1=py36_2
sip=4.19.12=py36h6538335_0
six=1.11.0=py36_1
snowballstemmer=1.2.1=py36h763602f_0
sphinx=1.7.9=py36_0
sphinxcontrib=1.0=py36_1
sphinxcontrib-websupport=1.1.0=py36_1
spyder=3.3.1=py36_1
spyder-kernels=0.2.6=py36_0
sqlite=3.24.0=h7602738_0
testpath=0.3.1=py36h2698cfe_0
tornado=5.1=py36hfa6e2cd_0
traitlets=4.3.2=py36h096827d_0
typed-ast=1.1.0=py36hfa6e2cd_0
urllib3=1.23=py36_0
vc=14=h0510ff6_3
vs2015_runtime=14.0.25123=3
wcwidth=0.1.7=py36h3d5aa90_0
webencodings=0.5.1=py36_1
wheel=0.31.1=py36_0
win_inet_pton=1.0.1=py36_1
wincertstore=0.2=py36h7fe50ca_0
wrapt=1.10.11=py36hfa6e2cd_2
zeromq=4.2.5=he025d50_1
zlib=1.2.11=h8395fce_2
```

**Comment by jacquerie** [comment]
Here's a reduction of the above code that still exhibits the bug:
```
In [1]: import pydot_ng as pydot

In [2]: pydot.find_graphviz()
Out[2]: 
{'circo': '/usr/bin/circo',
 'dot': '/usr/bin/dot',
 'fdp': '/usr/bin/fdp',
 'neato': '/usr/bin/neato',
 'sfdp': '/usr/bin/sfdp',
 'twopi': '/usr/bin/twopi'}

In [3]: from keras.models import Sequential

In [4]: from keras.layers import Embedding

In [5]: model = Sequential()

In [6]: model.add(Embedding(10000, 256, input_length=400, name='embed'))

In [7]: from keras.utils.vis_utils import plot_model

In [8]: plot_model(model, to_file='deployed_model.png', show_shapes=True, show_layer_names=True)
```
I also tested that replacing the `Embedding` layer with other layers (I tried `Dense` and `Dropout`) doesn't exhibit the bug.

**Comment by jacquerie** [comment]
I think that this bug is caused by a "hack" in Keras' implementation of the `Embedding` layer when used as the input layer: it actually creates two engine nodes, the first dealing with the input and the second actually doing the embedding. Because of this implementation detail the `_inbound_nodes` of the `Embedding` layer actually contains something, so the code goes through this loop
https://github.com/keras-team/keras/blob/590aa832680641218b233b22fa8b3bd7fdc0287e/keras/utils/vis_utils.py#L106-L111
in which an incoming edge is added to the `Embedding` layer. In fact, the "large number" that is then printed out is nothing but the `id` of the Python object corresponding to the engine node dealing with the input.

**Comment by jacquerie** [comment]
The above explanation identified the wrong root cause for this issue. See instead https://github.com/keras-team/keras/pull/11412#issue-223300903 for what I believe to be the root cause.
